### PR TITLE
Add native events types to DropDownEditor-/TextEditor-like widgets

### PR DESCRIPTION
**Pull request by pavelgruba**

### CODE REVIEW

File: js/ui/autocomplete.d.ts [hunk 1]
```diff
@@ -33,40 +33,40 @@ export type ClosedEvent = EventInfo<dxAutocomplete>;
...
 /** @public */
-export type FocusOutEvent = NativeEventInfo<dxAutocomplete>;
+export type FocusOutEvent = NativeEventInfo<dxAutocomplete, FocusEvent>;
 
 /** @public */
 export type InitializedEvent = InitializedEventInfo<dxAutocomplete>;
 
 /** @public */
-export type InputEvent = NativeEventInfo<dxAutocomplete>;
+export type InputEvent = NativeEventInfo<dxAutocomplete, UIEvent>;
```
Comment (DokaRus) [inline js/ui/autocomplete.d.ts:57]:
```suggestion
export type InputEvent = NativeEventInfo<dxAutocomplete, InputEvent>;
```
?

File: js/ui/autocomplete.d.ts [hunk 2]
```diff
@@ -75,13 +75,13 @@ export type OpenedEvent = EventInfo<dxAutocomplete>;
...
 /** @public */
-export type PasteEvent = NativeEventInfo<dxAutocomplete>;
+export type PasteEvent = NativeEventInfo<dxAutocomplete, ClipboardEvent>;
 
 /** @public */
 export type SelectionChangedEvent = EventInfo<dxAutocomplete> & SelectionChangedInfo;
 
 /** @public */
-export type ValueChangedEvent = NativeEventInfo<dxAutocomplete> & ValueChangedInfo;
+export type ValueChangedEvent = NativeEventInfo<dxAutocomplete, KeyboardEvent | MouseEvent | PointerEvent | UIEvent> & ValueChangedInfo;
```
Comment (DokaRus) [inline js/ui/autocomplete.d.ts]:
```suggestion
export type ValueChangedEvent = NativeEventInfo<dxAutocomplete, KeyboardEvent | MouseEvent | PointerEvent | Event> & ValueChangedInfo;
```

For example, [change](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event): `ChangeEvent: Event`

Depends on [valueChangeEvent ](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxAutocomplete/Configuration/#valueChangeEvent)option

File: js/ui/color_box.d.ts [hunk 3]
```diff
@@ -28,43 +28,43 @@ import {
...
 /** @public */
-export type FocusOutEvent = NativeEventInfo<dxColorBox>;
+export type FocusOutEvent = NativeEventInfo<dxColorBox, FocusEvent>;
 
 /** @public */
 export type InitializedEvent = InitializedEventInfo<dxColorBox>;
 
 /** @public */
-export type InputEvent = NativeEventInfo<dxColorBox>;
+export type InputEvent = NativeEventInfo<dxColorBox, UIEvent>;
```
Comment (DokaRus) [inline js/ui/color_box.d.ts:58]:
```suggestion
export type InputEvent = NativeEventInfo<dxColorBox, InputEvent>;
```

File: js/ui/date_box.d.ts [hunk 4]
```diff
@@ -37,37 +37,37 @@ export type ClosedEvent = EventInfo<dxDateBox>;
...
 /** @public */
-export type FocusOutEvent = NativeEventInfo<dxDateBox>;
+export type FocusOutEvent = NativeEventInfo<dxDateBox, FocusEvent>;
 
 /** @public */
 export type InitializedEvent = InitializedEventInfo<dxDateBox>;
 
 /** @public */
-export type InputEvent = NativeEventInfo<dxDateBox>;
+export type InputEvent = NativeEventInfo<dxDateBox, UIEvent>;
```
Comment (DokaRus) [inline js/ui/date_box.d.ts:61]:
```suggestion
export type InputEvent = NativeEventInfo<dxDateBox, InputEvent>;
```

File: js/ui/date_box.d.ts [hunk 5]
```diff
@@ -76,10 +76,10 @@ export type OpenedEvent = EventInfo<dxDateBox>;
 export type OptionChangedEvent = EventInfo<dxDateBox> & ChangedOptionInfo;
 
 /** @public */
-export type PasteEvent = NativeEventInfo<dxDateBox>;
+export type PasteEvent = NativeEventInfo<dxDateBox, ClipboardEvent>;
 
 /** @public */
-export type ValueChangedEvent = NativeEventInfo<dxDateBox> & ValueChangedInfo;
+export type ValueChangedEvent = NativeEventInfo<dxDateBox, KeyboardEvent | MouseEvent | PointerEvent | UIEvent> & ValueChangedInfo;
```
Comment (DokaRus) [inline js/ui/date_box.d.ts]:
```suggestion
export type ValueChangedEvent = NativeEventInfo<dxDateBox, KeyboardEvent | MouseEvent | PointerEvent | Event> & ValueChangedInfo;
```

https://github.com/DevExpress/DevExtreme/pull/19950/files#r741000230

File: js/ui/drop_down_box.d.ts [hunk 6]
```diff
@@ -34,43 +34,43 @@ import {
...
 /** @public */
-export type FocusOutEvent = NativeEventInfo<dxDropDownBox>;
+export type FocusOutEvent = NativeEventInfo<dxDropDownBox, FocusEvent>;
 
 /** @public */
 export type InitializedEvent = InitializedEventInfo<dxDropDownBox>;
 
 /** @public */
-export type InputEvent = NativeEventInfo<dxDropDownBox>;
+export type InputEvent = NativeEventInfo<dxDropDownBox, UIEvent>;
```
Comment (DokaRus) [inline js/ui/drop_down_box.d.ts:64]:
```suggestion
export type InputEvent = NativeEventInfo<dxDropDownBox, InputEvent>;
```

File: js/ui/drop_down_box.d.ts [hunk 7]
```diff
@@ -79,10 +79,10 @@ export type OpenedEvent = EventInfo<dxDropDownBox>;
 export type OptionChangedEvent = EventInfo<dxDropDownBox> & ChangedOptionInfo;
 
 /** @public */
-export type PasteEvent = NativeEventInfo<dxDropDownBox>;
+export type PasteEvent = NativeEventInfo<dxDropDownBox, ClipboardEvent>;
 
 /** @public */
-export type ValueChangedEvent = NativeEventInfo<dxDropDownBox> & ValueChangedInfo;
+export type ValueChangedEvent = NativeEventInfo<dxDropDownBox, KeyboardEvent | MouseEvent | PointerEvent | UIEvent> & ValueChangedInfo;
```
Comment (DokaRus) [inline js/ui/drop_down_box.d.ts]:
```suggestion
export type ValueChangedEvent = NativeEventInfo<dxDropDownBox, KeyboardEvent | MouseEvent | PointerEvent | Event> & ValueChangedInfo;
```
https://github.com/DevExpress/DevExtreme/pull/19950/files#r741000230

File: js/ui/drop_down_editor/ui.drop_down_list.d.ts [hunk 8]
```diff
@@ -107,7 +107,7 @@ export interface dxDropDownListOptions<TComponent> extends DataExpressionMixinOp
      * @action
      * @public
      */
-    onValueChanged?: ((e: NativeEventInfo<TComponent> & ValueChangedInfo) => void);
+    onValueChanged?: ((e: NativeEventInfo<TComponent, KeyboardEvent | MouseEvent | PointerEvent | UIEvent> & ValueChangedInfo) => void);
```
Comment (DokaRus) [inline js/ui/drop_down_editor/ui.drop_down_list.d.ts]:
```suggestion
    onValueChanged?: ((e: NativeEventInfo<TComponent, KeyboardEvent | MouseEvent | PointerEvent | Event> & ValueChangedInfo) => void);
```

https://github.com/DevExpress/DevExtreme/pull/19950/files#r741000230

File: js/ui/lookup.d.ts [hunk 9]
```diff
@@ -76,7 +76,7 @@ export type SelectionChangedEvent = EventInfo<dxLookup> & SelectionChangedInfo;
 export type TitleRenderedEvent = EventInfo<dxLookup> & TitleRenderedInfo;
 
 /** @public */
-export type ValueChangedEvent = NativeEventInfo<dxLookup> & ValueChangedInfo;
+export type ValueChangedEvent = NativeEventInfo<dxLookup, KeyboardEvent | MouseEvent | PointerEvent | UIEvent> & ValueChangedInfo;
```
Comment (DokaRus) [inline js/ui/lookup.d.ts]:
```suggestion
export type ValueChangedEvent = NativeEventInfo<dxLookup, KeyboardEvent | MouseEvent | PointerEvent | Event> & ValueChangedInfo;
```

https://github.com/DevExpress/DevExtreme/pull/19950/files#r741000230

File: js/ui/number_box.d.ts [hunk 10]
```diff
@@ -19,52 +19,52 @@ import {
...
 /** @public */
-export type FocusOutEvent = NativeEventInfo<dxNumberBox>;
+export type FocusOutEvent = NativeEventInfo<dxNumberBox, FocusEvent>;
 
 /** @public */
 export type InitializedEvent = InitializedEventInfo<dxNumberBox>;
 
 /** @public */
-export type InputEvent = NativeEventInfo<dxNumberBox>;
+export type InputEvent = NativeEventInfo<dxNumberBox, UIEvent>;
```
Comment (DokaRus) [inline js/ui/number_box.d.ts:49]:
```suggestion
export type InputEvent = NativeEventInfo<dxNumberBox, InputEvent>;
```

File: js/ui/number_box.d.ts [hunk 11]
```diff
@@ -19,52 +19,52 @@ import {
...
 /** @public */
 export type OptionChangedEvent = EventInfo<dxNumberBox> & ChangedOptionInfo;
 
 /** @public */
-export type PasteEvent = NativeEventInfo<dxNumberBox>;
+export type PasteEvent = NativeEventInfo<dxNumberBox, ClipboardEvent>;
 
 /** @public */
-export type ValueChangedEvent = NativeEventInfo<dxNumberBox> & ValueChangedInfo;
+export type ValueChangedEvent = NativeEventInfo<dxNumberBox, KeyboardEvent | MouseEvent | PointerEvent | TouchEvent | UIEvent | Event> & ValueChangedInfo;
```
Comment (DokaRus) [inline js/ui/number_box.d.ts]:
```suggestion
export type ValueChangedEvent = NativeEventInfo<dxNumberBox, KeyboardEvent | MouseEvent | PointerEvent | TouchEvent | Event> & ValueChangedInfo;
```
`UIEvent: Event`
Let's stay a base interface?

File: js/ui/select_box.d.ts [hunk 12]
```diff
@@ -47,42 +47,42 @@ export type ClosedEvent = EventInfo<dxSelectBox>;
...
 /** @public */
-export type FocusOutEvent = NativeEventInfo<dxSelectBox>;
+export type FocusOutEvent = NativeEventInfo<dxSelectBox, FocusEvent>;
 
 /** @public */
 export type InitializedEvent = InitializedEventInfo<dxSelectBox>;
 
 /** @public */
-export type InputEvent = NativeEventInfo<dxSelectBox>;
+export type InputEvent = NativeEventInfo<dxSelectBox, UIEvent>;
```
Comment (DokaRus) [inline js/ui/select_box.d.ts:73]:
```suggestion
export type InputEvent = NativeEventInfo<dxSelectBox, InputEvent>;
```

File: js/ui/select_box.d.ts [hunk 13]
```diff
@@ -91,13 +91,13 @@ export type OpenedEvent = EventInfo<dxSelectBox>;
...
 /** @public */
-export type PasteEvent = NativeEventInfo<dxSelectBox>;
+export type PasteEvent = NativeEventInfo<dxSelectBox, ClipboardEvent>;
 
 /** @public */
 export type SelectionChangedEvent = EventInfo<dxSelectBox> & SelectionChangedInfo;
 
 /** @public */
-export type ValueChangedEvent = NativeEventInfo<dxSelectBox> & ValueChangedInfo;
+export type ValueChangedEvent = NativeEventInfo<dxSelectBox, KeyboardEvent | MouseEvent | UIEvent> & ValueChangedInfo;
```
Comment (DokaRus) [inline js/ui/select_box.d.ts]:
```suggestion
export type ValueChangedEvent = NativeEventInfo<dxSelectBox, KeyboardEvent | MouseEvent | Event> & ValueChangedInfo;
```
https://github.com/DevExpress/DevExtreme/pull/19950/files#r741000230

File: js/ui/tag_box.d.ts [hunk 14]
```diff
@@ -49,31 +49,31 @@ export type CustomItemCreatingEvent = EventInfo<dxTagBox> & CustomItemCreatingIn
...
 /** @public */
-export type FocusOutEvent = NativeEventInfo<dxTagBox>;
+export type FocusOutEvent = NativeEventInfo<dxTagBox, FocusEvent>;
 
 /** @public */
 export type InitializedEvent = InitializedEventInfo<dxTagBox>;
 
 /** @public */
-export type InputEvent = NativeEventInfo<dxTagBox>;
+export type InputEvent = NativeEventInfo<dxTagBox, UIEvent>;
```
Comment (DokaRus) [inline js/ui/tag_box.d.ts:64]:
```suggestion
export type InputEvent = NativeEventInfo<dxTagBox, InputEvent>;
```

File: js/ui/tag_box.d.ts [hunk 15]
```diff
@@ -97,7 +97,7 @@ export type SelectAllValueChangedEvent = EventInfo<dxTagBox> & {
 export type SelectionChangedEvent = EventInfo<dxTagBox> & SelectionChangedInfo<string | number | any>;
 
 /** @public */
-export type ValueChangedEvent = NativeEventInfo<dxTagBox> & ValueChangedInfo;
+export type ValueChangedEvent = NativeEventInfo<dxTagBox, KeyboardEvent | MouseEvent | PointerEvent | UIEvent | Event> & ValueChangedInfo;
```
Comment (DokaRus) [inline js/ui/tag_box.d.ts]:
```suggestion
export type ValueChangedEvent = NativeEventInfo<dxTagBox, KeyboardEvent | MouseEvent | PointerEvent | Event> & ValueChangedInfo;
```

File: js/ui/text_area.d.ts [hunk 16]
```diff
@@ -14,52 +14,52 @@ import dxTextBox, {
...
 /** @public */
-export type FocusOutEvent = NativeEventInfo<dxTextArea>;
+export type FocusOutEvent = NativeEventInfo<dxTextArea, FocusEvent>;
 
 /** @public */
 export type InitializedEvent = InitializedEventInfo<dxTextArea>;
 
 /** @public */
-export type InputEvent = NativeEventInfo<dxTextArea>;
+export type InputEvent = NativeEventInfo<dxTextArea, UIEvent>;
```
Comment (DokaRus) [inline js/ui/text_area.d.ts:44]:
```suggestion
export type InputEvent = NativeEventInfo<dxTextArea, InputEvent>;
```

File: js/ui/text_area.d.ts [hunk 17]
```diff
@@ -14,52 +14,52 @@ import dxTextBox, {
...
 /** @public */
 export type OptionChangedEvent = EventInfo<dxTextArea> & ChangedOptionInfo;
 
 /** @public */
-export type PasteEvent = NativeEventInfo<dxTextArea>;
+export type PasteEvent = NativeEventInfo<dxTextArea, ClipboardEvent>;
 
 /** @public */
-export type ValueChangedEvent = NativeEventInfo<dxTextArea> & ValueChangedInfo;
+export type ValueChangedEvent = NativeEventInfo<dxTextArea, KeyboardEvent | MouseEvent | PointerEvent | TouchEvent | UIEvent | Event> & ValueChangedInfo;
```
Comment (DokaRus) [inline js/ui/text_area.d.ts]:
```suggestion
export type ValueChangedEvent = NativeEventInfo<dxTextArea, KeyboardEvent | MouseEvent | PointerEvent | TouchEvent | Event> & ValueChangedInfo;
```

File: js/ui/text_box.d.ts [hunk 18]
```diff
@@ -14,52 +14,52 @@ import dxTextEditor, {
...
 /** @public */
-export type FocusOutEvent = NativeEventInfo<dxTextBox>;
+export type FocusOutEvent = NativeEventInfo<dxTextBox, FocusEvent>;
 
 /** @public */
 export type InitializedEvent = InitializedEventInfo<dxTextBox>;
 
 /** @public */
-export type InputEvent = NativeEventInfo<dxTextBox>;
+export type InputEvent = NativeEventInfo<dxTextBox, UIEvent>;
```
Comment (DokaRus) [inline js/ui/text_box.d.ts:44]:
```suggestion
export type InputEvent = NativeEventInfo<dxTextBox, InputEvent>;
```

File: js/ui/text_box.d.ts [hunk 19]
```diff
@@ -14,52 +14,52 @@ import dxTextEditor, {
...
 /** @public */
 export type OptionChangedEvent = EventInfo<dxTextBox> & ChangedOptionInfo;
 
 /** @public */
-export type PasteEvent = NativeEventInfo<dxTextBox>;
+export type PasteEvent = NativeEventInfo<dxTextBox, ClipboardEvent>;
 
 /** @public */
-export type ValueChangedEvent = NativeEventInfo<dxTextBox> & ValueChangedInfo;
+export type ValueChangedEvent = NativeEventInfo<dxTextBox, KeyboardEvent | MouseEvent | PointerEvent | TouchEvent | UIEvent | Event> & ValueChangedInfo;
```
Comment (DokaRus) [inline js/ui/text_box.d.ts]:
```suggestion
export type ValueChangedEvent = NativeEventInfo<dxTextBox, KeyboardEvent | MouseEvent | PointerEvent | TouchEvent | Event> & ValueChangedInfo;
```

File: js/ui/text_box/ui.text_editor.base.d.ts [hunk 20]
```diff
@@ -164,7 +164,7 @@ export interface dxTextEditorOptions<TComponent> extends EditorOptions<TComponen
      * @action
      * @public
      */
-    onInput?: ((e: NativeEventInfo<TComponent>) => void);
+    onInput?: ((e: NativeEventInfo<TComponent, UIEvent>) => void);
```
Comment (DokaRus) [inline js/ui/text_box/ui.text_editor.base.d.ts:167]:
```suggestion
    onInput?: ((e: NativeEventInfo<TComponent, InputEvent>) => void);
```

**Review by DokaRus — DISMISSED** [review]

**Review by DokaRus — APPROVED** [review]

**Review by IlyaKhD — DISMISSED** [review]

**Pull request merged**